src/app/state/importaccounts/saga.ts

```diff
@@ -14,24 +14,58 @@ import type Transport from '@ledgerhq/hw-transport'
 import {
   selectImportAccountHasMissingBalances,
   selectImportAccounts,
-  selectImportAccountsOnCurrentPage,
   selectImportAccountsFullList,
+  selectImportAccountsOnCurrentPage,
   selectImportAccountsPageNumber,
+  selectSelectedBleDevice,
 } from './selectors'
 import { getAccountBalanceWithFallback } from '../../lib/getAccountBalanceWithFallback'
+import BleTransport from '@oasisprotocol/ionic-ledger-hw-transport-ble/lib'
+import { ScanResult } from '@capacitor-community/bluetooth-le'
 
 function* setStep(step: ImportAccountsStep) {
   yield* put(importAccountsActions.setStep(step))
 }
 
+function* isBluetoothSupported() {
+  const isSupported = yield* call([BleTransport, BleTransport.isSupported])
+  // https://github.com/WebBluetoothCG/web-bluetooth/blob/main/implementation-status.md#scanning-api
+  // !navigator.bluetooth
+  if (!isSupported) {
+    throw new WalletError(WalletErrors.BluetoothTransportNotSupported, 'BleTransport unsupported')
+  }
+}
+
+function* getBluetoothDevices() {
+  yield* call(isBluetoothSupported)
+  return yield* call(BleTransport.list)
+}
+
+function* getBluetoothTransport(device?: ScanResult) {
+  yield* call(isBluetoothSupported)
+
+  if (!device) {
+    throw new WalletError(WalletErrors.LedgerNoDeviceSelected, 'No device selected')
+  }
+
+  try {
+    return yield* call(BleTransport.open, device)
```

Comment:
Writing to device is not awaited:
https://github.com/oasisprotocol/ionic-ledger-hw-transport-ble/blob/9e3f0752db204241073da97e2708cad00724d976/src/index.ts#L341

https://github.com/oasisprotocol/ionic-ledger-hw-transport-ble/blob/9e3f0752db204241073da97e2708cad00724d976/src/index.ts#L348

Reply:
Nice catch 👍 Resolved here -> https://github.com/oasisprotocol/ionic-ledger-hw-transport-ble/pull/10